### PR TITLE
Allow sharing of private files 

### LINKED ISSUE #124: Allow sharing of private files
## Is your feature request related to a problem? Please describe.
I'm always frustrated when I follow the default path and then can't solve some trivial problem. So I can create a "private" container (which is the default) and then I would like to share it with my friend, which is rather typical. How would I do that?

## Describe the solution you'd like
Several possibilities:
 * special "share" link with embedded Bearer allowing to read for anyone who has this link
 * more convenient eACL rule manager allowing to specify an address to share object with (hi, nspcc-dev/neofs-api#278 though, likely not really possible at this stage)
 * a combination of those?

### PR DESCRIPTION
**Pull request by mike-petrov**
Signed-off-by: Mikhail Petrov <mike@nspcc.ru>

closes #124

### CODE REVIEW

**Comment by mike-petrov** [comment]
image

image

**Comment by mike-petrov** [comment]
image

**Comment by mike-petrov** [comment]
I noticed that access filter rules do not affect a specific object and give access to all objects in all containers with this token

File: src/App.js [hunk 1]
```diff
@@ -234,6 +246,23 @@ export const App = () => {
...
+		} else if (type === 'object' && params.objectId) {
+			body = [{
+				"object": [{
+					"operation": 'GET',
+					"action": "ALLOW",
+					"filters": [{
+						"headerType": "OBJECT",
+						"key": "$Object:objectID",
+						"matchType": "STRING_EQUAL",
+						"value": params.objectId,
```
Comment (mike-petrov) [inline src/App.js:258]: I noticed that access filter rules do not affect a specific object and give access to all objects in all containers with this token, what am I doing wrong here?

Comment (roman-khimov) [reply src/App.js:258]: You need an additional rule to forbid access for OTHERS right after the "good" one.

File: src/App.js [hunk 2]
```diff
@@ -407,28 +407,24 @@ export const App = () => {
 	};
 
 	const onDeleteContainer = (containerName) => {
-		if (walletData.tokens.container.DELETE) {
```
[the remark this answers is not in the document]
Comment (mike-petrov) [reply src/App.js:410]: we can't trigger this function without token, that's why it's useless:

image

File: src/App.js [hunk 3]
```diff
@@ -267,7 +341,7 @@ export const App = () => {
 		api('POST', '/auth', body, {
 			[ContentTypeHeader]: "application/json",
 			[BearerOwnerIdHeader]: walletData.account.address,
-			[BearerLifetime]: 2,
+			[BearerLifetime]: params.objectId ? 24 : 2,
```
Comment (roman-khimov) [inline src/App.js:344]: This limits lifetime to 24h, but how would user know about it or be able to change?

Comment (mike-petrov) [reply src/App.js:344]: image

Comment (mike-petrov) [reply src/App.js:344]: Now it's not customizable. Should we do it?

File: src/App.js [hunk 4]
```diff
@@ -234,6 +246,68 @@ export const App = () => {
...
+					}]
+				}, {
+					"operation": 'DELETE',
+					"action": "DENY",
+					"filters": [],
+					"targets": [{
+						"role": "OTHERS",
+						"keys": []
+					}]
+				}]
```
Comment (roman-khimov) [inline src/App.js:309]: Let's also add three rules forbidding `SEARCH, RANGE, RANGEHASH`. Token can be misused otherwise.

File: src/App.js (hunk 3 above)
Comment (roman-khimov) [reply src/App.js:344]: OK, let's have it this way for now and leave #175 for the future.

File: src/App.js [hunk 5]
```diff
@@ -234,6 +246,49 @@ export const App = () => {
...
+						"headerType": "OBJECT",
+						"key": "$Object:objectID",
+						"matchType": "STRING_EQUAL",
+						"value": params.objectId,
+					}],
+					"targets": [{
+						"role": "OTHERS",
+						"keys": []
+					}]
+				}, ...presets.forbid.eACLParams]
```
Comment (roman-khimov) [inline src/App.js:290]: Nice one!

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**